### PR TITLE
ftintitle: New customization option to keep feature in artist field

### PR DESCRIPTION
**Pull request by klb2**
## Description

This PR adds a new option `keep_in_artist` to the `ftintitle` plugin.

### Previous Behavior
The artist field is _always_ changed to match the albumartist field.  

### New Behavior
The new option allows a customization of the old behavior. If the option is set to yes/true, the "feat. X" part is kept in the artist field.
In order to be backwards compatible and not introduce unexpected behavior, the default is set to false, i.e., matching the old behavior.

### Benefits
- Additional customization
- Searching for features using both title and artist (this can be very helpful when using music players with GUI)

## To Do

- [X] Documentation.
- [X] Changelog.
- [X] Tests.

### CODE REVIEW

**Review by JOJ0 — CHANGES_REQUESTED** [review]
This looks very straight forward. Thanks! I think it doesn't hurt if you also add the typical usecase you mention in the PR description to the docs as well:

> Searching for features using both title and artist (this can be very helpful when using music players with GUI)

Either in the docs introductional text or in the option description itself.

Depending on where you put it, you might want to rephrase. Again just brainstorming here. Something like:

....To be able to search for features in artist fields as well, this option can be used.....

It's helpful in music players that let us search in artist fields in general, not only with GUI.

File: beetsplug/ftintitle.py [hunk 1]
```diff
@@ -120,15 +122,19 @@ def imported(self, session, task):
...
         metadata. Also, print out messages about any changes that are made.
         If `drop_feat` is set, then do not add the artist to the title; just
         remove it from the artist field.
         """
-        # In all cases, update the artist fields.
-        self._log.info("artist: {0} -> {1}", item.artist, item.albumartist)
-        item.artist = item.albumartist
+        # In case the artist is kept, do not update the artist fields.
+        if keep_in_artist_field:
+            self._log.info("artist: {0} -> {1}", item.artist, item.artist)
```
[the remark this answers is not in the document]
Comment (klb2) [reply beetsplug/ftintitle.py]: That is a good idea. I have updated the log message.

**Review by klb2 — COMMENTED** [review]
Looks good to me. Please let me know if you have other change requests

**Comment by klb2** [comment]
I have updated the documentation and log message as you have suggested. Please let me know if there are any other changes you would like to see.

File: beetsplug/ftintitle.py [hunk 2]
```diff
@@ -120,15 +122,21 @@ def imported(self, session, task):
...
         remove it from the artist field.
         """
-        # In all cases, update the artist fields.
-        self._log.info("artist: {0} -> {1}", item.artist, item.albumartist)
-        item.artist = item.albumartist
+        # In case the artist is kept, do not update the artist fields.
+        if keep_in_artist_field:
+            self._log.info(
+                "artist: {0} (Not changing due to keep_in_artist)", item.artist
+            )
```
Comment (JOJ0) [inline beetsplug/ftintitle.py:135]: 👍Perfect! Thanks!

File: docs/plugins/ftintitle.rst [hunk 3]
```diff
@@ -27,6 +27,10 @@ file. The available options are:
 - **format**: Defines the format for the featuring X  part of the new title field.
   In this format the ``{0}`` is used to define where the featured artists are placed.
   Default: ``feat. {0}``
+- **keep_in_artist**: Keep the featuring X part in the artist field. This can
+  be useful if you still want to be able to search for features in the artist
+  field.
+  Default: ``no``.
```
Comment (JOJ0) [inline docs/plugins/ftintitle.rst:33]: Well written. Straight to the point. I like! 😁

**Review by JOJ0 — APPROVED** [review]
Many thanks. This is good to go 🎉

**Comment by chrishoage** [comment]
This PR produces errors for me 

```
Traceback (most recent call last):
  File "/lsiopy/bin/beet", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/lsiopy/lib/python3.12/site-packages/beets/ui/__init__.py", line 1865, in main
    _raw_main(args)
  File "/lsiopy/lib/python3.12/site-packages/beets/ui/__init__.py", line 1852, in _raw_main
    subcommand.func(lib, suboptions, subargs)
  File "/lsiopy/lib/python3.12/site-packages/beets/ui/commands.py", line 1400, in import_func
    import_files(lib, paths, query)
  File "/lsiopy/lib/python3.12/site-packages/beets/ui/commands.py", line 1331, in import_files
    session.run()
  File "/lsiopy/lib/python3.12/site-packages/beets/importer.py", line 360, in run
    pl.run_parallel(QUEUE_SIZE)
  File "/lsiopy/lib/python3.12/site-packages/beets/util/pipeline.py", line 447, in run_parallel
    raise exc_info[1].with_traceback(exc_info[2])
  File "/lsiopy/lib/python3.12/site-packages/beets/util/pipeline.py", line 312, in run
    out = self.coro.send(msg)
          ^^^^^^^^^^^^^^^^^^^
  File "/lsiopy/lib/python3.12/site-packages/beets/util/pipeline.py", line 195, in coro
    func(*(args + (task,)))
  File "/lsiopy/lib/python3.12/site-packages/beets/importer.py", line 1668, in plugin_stage
    func(session, task)
  File "/lsiopy/lib/python3.12/site-packages/beets/plugins.py", line 143, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/lsiopy/lib/python3.12/site-packages/beetsplug/ftintitle.py", line 122, in imported
    self.ft_in_title(item, drop_feat)
TypeError: FtInTitlePlugin.ft_in_title() missing 1 required positional argument: 'keep_in_artist_field'
```

Line here https://github.com/beetbox/beets/blob/cd360b6a99bef5f8b5e4ffce334e89b580bbd652/beetsplug/ftintitle.py#L122

I am able to reproduce just by running `beets import -q /path/to/music`

**Comment by klb2** [comment]
Thank you for spotting this @chrishoage. I have created a PR to fix this (#5433)

**Pull request merged**